realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/KotlinSyncedRealmTests.kt

```diff
@@ -122,6 +122,7 @@ class KotlinSyncedRealmTests { // FIXME: Rename to SyncedRealmTests once remaini
         val config2 = createDefaultConfig(user2, partitionValue)
         realm = Realm.getInstance(config2)
         realm.syncSession.downloadAllServerChanges()
+        realm.refresh()
```

Comment:
Is this normal way of doing it. If so, seems like something others could also forget and it could be worth considering adding an argument to `downloadAllServerChanges` to force people to think of it.

Reply:
Yes, the problem is that `downloadAllServerChanges` block until the background thread doing the download is done, but the Realm on the caller thread is not automatically updated. I created this issue: https://github.com/realm/realm-java/issues/6814 to track it. Because it is not entirely trivial.